src/Files.App/ViewModels/SidebarViewModel.cs

```diff
@@ -6,230 +6,133 @@
-			set
-			{
-				if (value == UserSettingsService.AppearanceSettingsService.ShowWslSection) 
-					return;
-
-				UserSettingsService.AppearanceSettingsService.ShowWslSection = value;
-			}
+			get => userSettingsService.AppearanceSettingsService.ShowWslSection;
+			set => userSettingsService.AppearanceSettingsService.ShowWslSection = value;
 		}
-
 		public bool ShowFileTagsSection
 		{
-			get => UserSettingsService.AppearanceSettingsService.ShowFileTagsSection;
-			set
-			{
-				if (value == UserSettingsService.AppearanceSettingsService.ShowFileTagsSection) 
-					return;
-
-				UserSettingsService.AppearanceSettingsService.ShowFileTagsSection = value;
-			}
-		}
-
-		private INavigationControlItem selectedSidebarItem;
-
-		public INavigationControlItem SidebarSelectedItem
-		{
-			get => selectedSidebarItem;
-			set => SetProperty(ref selectedSidebarItem, value);
+			get => userSettingsService.AppearanceSettingsService.ShowFileTagsSection;
+			set => userSettingsService.AppearanceSettingsService.ShowFileTagsSection = value;
 		}
 
 		public SidebarViewModel()
 		{
 			dispatcherQueue = DispatcherQueue.GetForCurrentThread();
 
 			SideBarItems = new BulkConcurrentObservableCollection<INavigationControlItem>();
-			EmptyRecycleBinCommand = new RelayCommand<RoutedEventArgs>(EmptyRecycleBin);
-			UserSettingsService.OnSettingChangedEvent += UserSettingsService_OnSettingChangedEvent;
+			EmptyRecycleBinCommand = new AsyncRelayCommand(RecycleBinHelpers.S_EmptyRecycleBin);
+			userSettingsService.OnSettingChangedEvent += UserSettingsService_OnSettingChangedEvent;
 			CreateItemHome();
 
-			Manager_DataChanged(SectionType.Favorites, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-			Manager_DataChanged(SectionType.Library, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-			Manager_DataChanged(SectionType.Drives, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-			Manager_DataChanged(SectionType.CloudDrives, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-			Manager_DataChanged(SectionType.Network, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-			Manager_DataChanged(SectionType.WSL, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
-			Manager_DataChanged(SectionType.FileTag, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
+			var resetEventArgs = new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset);
+			Manager_DataChanged(SectionType.Favorites, resetEventArgs);
+			Manager_DataChanged(SectionType.Library, resetEventArgs);
+			Manager_DataChanged(SectionType.Drives, resetEventArgs);
+			Manager_DataChanged(SectionType.CloudDrives, resetEventArgs);
+			Manager_DataChanged(SectionType.Network, resetEventArgs);
+			Manager_DataChanged(SectionType.WSL, resetEventArgs);
+			Manager_DataChanged(SectionType.FileTag, resetEventArgs);
 
```

Comment:
Can all the `new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset` be safely represented by one object where they used to be different objects before?

Reply:
Yes because it is read-only and the NewItems and OldItems properties are null.